perception/traffic_light_multi_camera_fusion/src/node.cpp

```diff
@@ -0,0 +1,291 @@
+// Copyright 2023 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "traffic_light_multi_camera_fusion/node.hpp"
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace
+{
+
+bool isUnknown(const autoware_auto_perception_msgs::msg::TrafficSignal & signal)
+{
+  return signal.lights.size() == 1 &&
+         signal.lights[0].color == autoware_auto_perception_msgs::msg::TrafficLight::UNKNOWN &&
+         signal.lights[0].shape == autoware_auto_perception_msgs::msg::TrafficLight::UNKNOWN;
+}
+
+/**
+ * @brief Currently the visible score only considers the truncation.
+ * If the detection roi is very close to the image boundary, it would be considered as truncated.
+ *
+ * @param roi       detection roi
+ * @param cam_info  camera info
+ * @return 0 if traffic light is truncated, otherwise 1
+ */
+int calVisibleScore(
+  const traffic_light::MultiCameraFusion::RoiType & roi,
+  const traffic_light::MultiCameraFusion::CamInfoType & cam_info)
+{
+  uint32_t boundary = 5;
```

Comment:
@Mingyu1991 
nitpick: magic numbers could be added as parameters or constexpr/const

Reply:
Modified